python-avd/pyavd/_eos_designs/structured_config/underlay/dhcp_servers.py

```diff
@@ -69,19 +70,34 @@ def _update_subnets(self: AvdStructuredConfigUnderlayProtocol, dhcp_server: EosC
                     subnet_item.ranges.append_new(start=str(uplink.ip_address), end=str(uplink.ip_address))
                     dhcp_server.subnets.append(subnet_item)
 
+    def _get_cvp_server_for_dhcp(self: AvdStructuredConfigUnderlayProtocol) -> str | None:
+        """Return the first CVP server using either new or old data models."""
+        if self.inputs.cv_settings.cvaas.enabled:
+            region = next(iter(self.inputs.cv_settings.cvaas.clusters)).region
+            return CV_REGION_TO_SERVER_MAP[region]
+
+        if self.inputs.cv_settings.onprem_clusters:
+            return next(iter(self.inputs.cv_settings.onprem_clusters)).name
```

Comment:
Name of the on-prem cluster here may not be DNS-resolvable, routable or responding to HTTP(s). Option 67 should be instead pointing to the FQDN/IP of the first node of the cluster
```suggestion
            return next(iter(next(iter(self.inputs.cv_settings.onprem_clusters)).servers)).name
```

Reply:
absolutely. Thank you for catching.